src/AspNetCoreModuleV2/AspNetCore/src/applicationinfo.cpp

```diff
@@ -443,14 +406,10 @@ APPLICATION_INFO::FindNativeAssemblyFromHostfxr(
     {
         // Host fxr version is incorrect (need a higher version).
         // TODO log error
-        hr = E_FAIL;
-        goto Finished;
+        FINISHED(E_FAIL);
```

Comment:
FINISHED doesn't imply it failed. FINISHED_FAIL?

Reply:
FINISHED is more like unconditional return/